client/src/main/scala/org/lyranthe/prometheus/client/scala_syntax.scala

```diff
@@ -0,0 +1,84 @@
+package org.lyranthe.prometheus.client
+
+import org.lyranthe.prometheus.client.counter.LabelledCounter
+import org.lyranthe.prometheus.client.histogram.LabelledHistogram
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.{Failure, Success, Try}
+
+object scala_syntax {
+
+  object helper {
+
+    def all[T, U](metric: U): Try[T] => Option[U] = {
+      _ => Some(metric)
+    }
+
+    def success[T, U](metric: U): Try[T] => Option[U] = {
+      case Success(_) => Some(metric)
+      case Failure(_) => None
+    }
+
+    def failure[T, U](metric: U): Try[T] => Option[U] = {
+      case Success(_) => None
+      case Failure(_) => Some(metric)
+    }
+
+  }
+
+  def count[T](counter: LabelledCounter)(t: => Try[T]): Try[T] = {
+    countN(t)(helper.all(counter))
+  }
+
+  def countSuccess[T](counter: LabelledCounter)(t: => Try[T]): Try[T] = {
+    countN(t)(helper.success(counter))
+  }
+
+  def countFailure[T](counter: LabelledCounter)(t: => Try[T]): Try[T] = {
+    countN(t)(helper.failure(counter))
+  }
+
+  def countN[T](t: => Try[T])(f: (Try[T] => Option[LabelledCounter])*): Try[T] = {
+    val attempt = t
```

Comment:
So far as counting doesn't mind when a value is initialized, it doesn't matter (it matters for timing, not counting).